src/www/ui/api/Controllers/UploadTreeController.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/*
+ SPDX-FileCopyrightText: © 2023 Samuel Dushimimana <dushsam100@gmail.com>
+
+ SPDX-License-Identifier: GPL-2.0-only
+*/
+
+/**
+ * @file
+ * @brief Controller for uploadtree queries
+ */
+
+namespace Fossology\UI\Api\Controllers;
+
+use Fossology\UI\Api\Helper\ResponseHelper;
+use Fossology\UI\Api\Models\Info;
+use Fossology\UI\Api\Models\InfoType;
+use Psr\Http\Message\ServerRequestInterface;
+
+
+/**
+ * @class UploadTreeController
+ * @brief Controller for UploadTree model
+ */
+class UploadTreeController extends RestController
+{
+
+  /**
+   * Get the contents of a specific file
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function viewLicenseFile($request, $response, $args)
+  {
+    $upload_pk = intval($args['id']);
+    $item_pk = intval($args['itemId']);
+    $_GET['upload'] = $upload_pk;
+    $_GET['item'] = $item_pk;
+
+    $uploadDao = $this->restHelper->getUploadDao();
+    $returnVal = null;
+
+    if (!$this->dbHelper->doesIdExist("upload", "upload_pk", $upload_pk)) {
+      $returnVal = new Info(404, "Upload does not exist", InfoType::ERROR);
+    } else if (!$this->dbHelper->doesIdExist($uploadDao->getUploadtreeTableName($upload_pk), "uploadtree_pk", $item_pk)) {
+      $returnVal = new Info(404, "Item does not exist", InfoType::ERROR);
+    }
+
+    if ($returnVal !== null) {
+      return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+    }
+
+    $view = $this->restHelper->getPlugin('view');
+
+    list($pageMenu, $returnVal) = $view->getView(NULL, "license", 0, "", array(), false, true);
+    return $response->withJson(strip_tags($returnVal), 200);
```

Comment:
Also, would suggest to implement a similar function as `getView()` (or break it into two) with all the validation in place and return the raw file content. As the function is currently adding HTML tags to the response and using `strip_tags()` is dodgy as the file content itself can be in HTML.